fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/NestedRecordTypeBuilder.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * NestedRecordTypeBuilder.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2023 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.metadata;
+
+import com.apple.foundationdb.annotation.API;
+import com.google.protobuf.Descriptors;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Builder class for {@link NestedRecordType}s. This is used primarily by the {@link UnnestedRecordTypeBuilder} to keep
+ * track of nested types.
+ */
+@API(API.Status.EXPERIMENTAL)
+public class NestedRecordTypeBuilder extends RecordTypeBuilder {
+    public NestedRecordTypeBuilder(@Nonnull final Descriptors.Descriptor descriptor) {
+        super(descriptor);
+    }
+}
```

Comment:
Doesn't this need to override `build` to make the subclass? (Never called by `SyntheticRecordTypeBuilder`'s subclasses' `build` I guess.)